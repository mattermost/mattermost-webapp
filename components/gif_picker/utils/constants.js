// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    ItemTapAction: {
        OPEN_EMBED_PAGE: 1,
        SHARE: 2,
    },

    Tab: {
        TRENDING: 0,
        REACTIONS: 1,
    },

    appName: {
        mattermost: 'mattermost',
    },
};
