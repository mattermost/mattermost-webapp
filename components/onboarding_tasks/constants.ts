// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const FINISHED = 999;

export const OnboardingTaskCategory = 'onboarding_task_list';

// Whole task list is based on these
export const OnboardingTasksName = {
    CREATE_FROM_WORK_TEMPLATE: 'create_from_work_template',
    CHANNELS_TOUR: 'channels_tour',
    BOARDS_TOUR: 'boards_tour',
    PLAYBOOKS_TOUR: 'playbooks_tour',
    INVITE_PEOPLE: 'invite_people',
    DOWNLOAD_APP: 'download_app',
    COMPLETE_YOUR_PROFILE: 'complete_your_profile',
    EXPLORE_OTHER_TOOLS: 'explore_other_tools',
    VISIT_SYSTEM_CONSOLE: 'visit_system_console',
    START_TRIAL: 'start_trial',
};

export const OnboardingTaskList = {
    ONBOARDING_TASK_LIST_OPEN: 'onboarding_task_list_open',
    ONBOARDING_TASK_LIST_SHOW: 'onboarding_task_list_show',
    ONBOARDING_TASK_LIST_CLOSE: 'onboarding_task_list_close',
    ONBOARDING_VIDEO_MODAL: 'onboarding_video_modal',
    DECLINED_ONBOARDING_TASK_LIST: 'declined_onboarding_task_list',

    // auto show channel linked boards A/B test
    ONBOARDING_LINKED_BOARD_AUTO_SHOWN: 'linked_board_auto_shown',
};

export const GenericTaskSteps = {
    START: 0,
    STARTED: 1,
    FINISHED,
};

export const TaskNameMapToSteps = {
    [OnboardingTasksName.CREATE_FROM_WORK_TEMPLATE]: GenericTaskSteps,
    [OnboardingTasksName.CHANNELS_TOUR]: GenericTaskSteps,
    [OnboardingTasksName.BOARDS_TOUR]: GenericTaskSteps,
    [OnboardingTasksName.PLAYBOOKS_TOUR]: GenericTaskSteps,
    [OnboardingTasksName.COMPLETE_YOUR_PROFILE]: GenericTaskSteps,
    [OnboardingTasksName.EXPLORE_OTHER_TOOLS]: GenericTaskSteps,
    [OnboardingTasksName.DOWNLOAD_APP]: GenericTaskSteps,
    [OnboardingTasksName.VISIT_SYSTEM_CONSOLE]: GenericTaskSteps,
    [OnboardingTasksName.INVITE_PEOPLE]: GenericTaskSteps,
    [OnboardingTasksName.START_TRIAL]: GenericTaskSteps,
};

