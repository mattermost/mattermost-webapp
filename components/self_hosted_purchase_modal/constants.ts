// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const STORAGE_KEY_PURCHASE_IN_PROGRESS = 'PURCHASE_IN_PROGRESS';
