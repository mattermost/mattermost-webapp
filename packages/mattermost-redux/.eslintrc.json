{
  "rules": {
    "no-restricted-imports": ["error", {
      "patterns": [
        {"group": ["actions/*", "selectors/*", "types/*", "utils/*", "!utils/test_helper"], "message": "mattermost-redux can't import files from the rest of the web app"}
      ]
    }]
  }
}
