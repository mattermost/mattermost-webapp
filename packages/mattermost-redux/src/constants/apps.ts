// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {AppCallResponseType, AppCallType, AppExpandLevel, AppFieldType} from 'mattermost-redux/types/apps';

// This file's contents belong to the Apps Framework feature.
// Apps Framework feature is experimental, and the contents of this file are
// susceptible to breaking changes without pushing the major version of this package.

export const AppBindingLocations = {
    POST_MENU_ITEM: '/post_menu',
    CHANNEL_HEADER_ICON: '/channel_header',
    COMMAND: '/command',
    IN_POST: '/in_post',
};

export const AppBindingPresentations = {
    MODAL: 'modal',
};

export const AppCallResponseTypes: { [name: string]: AppCallResponseType } = {
    OK: 'ok',
    ERROR: 'error',
    FORM: 'form',
    CALL: 'call',
    NAVIGATE: 'navigate',
};

export const AppCallTypes: { [name: string]: AppCallType } = {
    SUBMIT: 'submit',
    LOOKUP: 'lookup',
    FORM: 'form',
    CANCEL: 'cancel',
};

export const AppExpandLevels: { [name: string]: AppExpandLevel } = {
    EXPAND_DEFAULT: '',
    EXPAND_NONE: 'none',
    EXPAND_ALL: 'all',
    EXPAND_SUMMARY: 'summary',
};

export const AppFieldTypes: { [name: string]: AppFieldType } = {
    TEXT: 'text',
    STATIC_SELECT: 'static_select',
    DYNAMIC_SELECT: 'dynamic_select',
    BOOL: 'bool',
    USER: 'user',
    CHANNEL: 'channel',
};
