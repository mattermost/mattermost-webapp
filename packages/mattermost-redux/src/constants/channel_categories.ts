// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ChannelCategoryType} from 'mattermost-redux/types/channel_categories';

export const CategoryTypes: {[name: string]: ChannelCategoryType} = {
    FAVORITES: 'favorites',
    CHANNELS: 'channels',
    DIRECT_MESSAGES: 'direct_messages',
    CUSTOM: 'custom',
};
