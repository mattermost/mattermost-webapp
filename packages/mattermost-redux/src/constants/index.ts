// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import General from './general';
import RequestStatus from './request_status';
import WebsocketEvents from './websocket';
import Preferences from './preferences';
import Posts from './posts';
import Files from './files';
import Teams from './teams';
import Stats from './stats';
import Permissions from './permissions';
import Emoji from './emoji';
import Plugins from './plugins';
import Groups from './groups';
import Users from './users';
import Roles from './roles';
import Threads from './threads';

export {General, Preferences, Posts, Files, RequestStatus, WebsocketEvents, Teams, Stats, Permissions, Emoji, Plugins, Groups, Users, Roles, Threads};
