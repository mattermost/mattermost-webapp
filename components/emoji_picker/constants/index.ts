// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory} from '@mattermost/types/emojis';

import * as Emoji from 'utils/emoji';

import {Category, Categories} from '../types';

export const RECENT = 'recent';
export const SEARCH_RESULTS = 'searchResults';
export const SMILEY_EMOTION = 'smileys-emotion';
export const CUSTOM = 'custom';

const categoryClass: Map<EmojiCategory, string> = new Map([
    [RECENT, 'icon-clock-outline'],
    [SMILEY_EMOTION, 'icon-emoticon-happy-outline'],
    ['people-body', 'icon-account-outline'],
    ['animals-nature', 'icon-leaf-outline'],
    ['food-drink', 'icon-food-apple'],
    ['activities', 'icon-basketball'],
    ['travel-places', 'icon-airplane-variant'],
    ['objects', 'icon-lightbulb-outline'],
    ['symbols', 'icon-heart-outline'],
    ['flags', 'icon-flag-outline'],
    [CUSTOM, 'icon-emoticon-custom-outline'],
    [SEARCH_RESULTS, ''],
]);

function createCategory(name: EmojiCategory): Category {
    return {
        name,
        id: Emoji.CategoryTranslations.get(name) || '',
        className: categoryClass.get(name) || '',
        message: Emoji.CategoryMessage.get(name)!,
    };
}

export const RECENT_EMOJI_CATEGORY: Pick<Categories, 'recent'> = {recent: createCategory(RECENT)};
export const SEARCH_EMOJI_CATEGORY: Pick<Categories, typeof SEARCH_RESULTS> = {searchResults: createCategory(SEARCH_RESULTS)};

export const CATEGORIES: Categories = Emoji.CategoryNames.
    filter((category) => !(category === 'recent' || category === 'searchResults')).
    reduce((previousCategory, currentCategory) => {
        return {
            ...previousCategory,
            [currentCategory]: createCategory(currentCategory as EmojiCategory),
        };
    }, {} as Categories);

export const EMOJI_PER_ROW = 9; // needs to match variable `$emoji-per-row` in _variables.scss
export const ITEM_HEIGHT = 36; //as per .emoji-picker__item height in _emoticons.scss
export const EMOJI_CONTAINER_HEIGHT = 290; // If this changes, the spaceRequiredAbove and spaceRequiredBelow props passed to the EmojiPickerOverlay must be updated

export const CATEGORY_HEADER_ROW = 'categoryHeaderRow';
export const EMOJIS_ROW = 'emojisRow';

export const EMOJI_SCROLL_THROTTLE_DELAY = 150;
export const EMOJI_ROWS_OVERSCAN_COUNT = 1;

export const CUSTOM_EMOJIS_PER_PAGE = 200;
export const CUSTOM_EMOJI_SEARCH_THROTTLE_TIME_MS = 1000;
