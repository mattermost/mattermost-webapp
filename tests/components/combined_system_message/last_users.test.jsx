// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Posts} from 'mattermost-redux/constants';

import LastUsers from 'components/post_view/combined_system_message/last_users.jsx';

describe('components/post_view/combined_system_message/LastUsers', () => {
    const baseProps = {
        actor: 'user_1',
        expandedLocale: {
            id: 'combined_system_message.added_to_channel.many_expanded',
            defaultMessage: '{users} and {lastUser} <b>added to the channel</b> by {actor}.',
        },
        postType: Posts.POST_TYPES.ADD_TO_CHANNEL,
        userDisplayNames: ['user_2', 'user_3', 'user_4 '],
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <LastUsers {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, expanded', () => {
        const wrapper = shallow(
            <LastUsers {...baseProps}/>
        );

        wrapper.setState({expand: true});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match state on handleOnClick', () => {
        const wrapper = shallow(
            <LastUsers {...baseProps}/>
        );

        wrapper.setState({expand: false});
        wrapper.instance().handleOnClick({preventDefault: jest.fn()});
        expect(wrapper.state('expand')).toBe(true);
    });
});
