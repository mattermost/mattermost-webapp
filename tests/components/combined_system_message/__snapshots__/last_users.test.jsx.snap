// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/combined_system_message/LastUsers should match snapshot 1`] = `
<React.Fragment>
  <FormattedMessage
    defaultMessage="{firstUser} and "
    id="last_users_message.first"
    values={
      Object {
        "firstUser": "user_2",
      }
    }
  />
  <a
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="{numOthers} others "
      id="last_users_message.others"
      values={
        Object {
          "numOthers": 2,
        }
      }
    />
  </a>
  <FormattedHTMLMessage
    defaultMessage="were <b>added to the channel</b> by {actor}."
    id="last_users_message.added_to_channel.type"
    values={
      Object {
        "actor": "user_1",
      }
    }
  />
</React.Fragment>
`;

exports[`components/post_view/combined_system_message/LastUsers should match snapshot, expanded 1`] = `
<FormattedHTMLMessage
  defaultMessage="{users} and {lastUser} <b>added to the channel</b> by {actor}."
  id="combined_system_message.added_to_channel.many_expanded"
  values={
    Object {
      "actor": "user_1",
      "lastUser": "user_4 ",
      "users": "user_2, user_3",
    }
  }
/>
`;
